### PR TITLE
Fix generation scripts for bwc indexes, and add 2.0 beta1 index

### PR DESCRIPTION
**Pull request by rjernst**
The scripts to download and build static bwc indexes had some issues that prevented them from working. They were trying to do things on 2.0 that were no longer supported. This change limits the old behavior to pre 2.0 index versions.

### CODE REVIEW

File: dev-tools/get-bwc-version.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def main():
     # 1.2.0 was pulled from download.elasticsearch.org because of routing bug:
     url = 'http://central.maven.org/maven2/org/elasticsearch/elasticsearch/1.2.0/%s' % filename
   else:
```
Comment (s1monw) [inline dev-tools/get-bwc-version.py]: this won't allow us to downlaod 1.x version and before shoudl we do the same thing here like:
```diff
-    url = 'https://download.elasticsearch.org/elasticsearch/elasticsearch/%s' % filename
+    if c.version.startswith('0') or c.version.startswith('1'):
+      url = 'https://download.elasticsearch.org/elasticsearch/elasticsearch/%s' % filename
+    else:
+      url = 'http://download.elasticsearch.org/elasticsearch/release/org/elasticsearch/distribution/tar/elasticsearch/%s/%s' % (c.version, filename)
+
```

**Comment by s1monw** [comment]
left one comment LGTM otherwise

**Comment by s1monw** [comment]
hmm I don't see the BWC index here are you adding this in a different PR?

File: core/src/test/java/org/elasticsearch/index/mapper/string/StringFieldMapperPositionIncrementGapTests.java [hunk 2]
```diff
@@ -51,7 +51,7 @@ public void testDefault() throws IOException {
      * Asserts that the post-2.0 default is being applied.
      */
     public static void assertGapIsOneHundred(Client client, String indexName, String type) throws IOException {
-        testGap(client(), indexName, type, 100);
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/index/mapper/string/StringFieldMapperPositionIncrementGapTests.java]: ++

**Comment by rjernst** [comment]
Thanks @s1monw I fixed the download script to still work with old indexes. This PR is just to make the scripts runnable again. I didn't want to tie that to updating the version constants.

**Comment by s1monw** [comment]
++ thx ryan

**Pull request merged**